com.unity.netcode.gameobjects/Prototyping/NetworkTransform.cs

```diff
@@ -377,22 +651,65 @@ private void FixedUpdate()
                 return;
             }
 
-            if (IsServer)
+            if (IsServer && UseFixedUpdate) // todo change IsServer for HasAuthority check
             {
                 // try to update local NetworkState
-                if (UpdateNetworkState(ref LocalNetworkState))
-                {
-                    // if updated (dirty), change NetVar, mark it dirty
-                    ReplNetworkState.Value = LocalNetworkState;
-                    ReplNetworkState.SetDirty(true);
-                }
+                DoUpdateToGhosts(NetworkManager.LocalTime.FixedTime);
             }
+
             // try to update previously consumed NetworkState
             // if we have any changes, that means made some updates locally
             // we apply the latest ReplNetworkState again to revert our changes
-            else if (UpdateNetworkState(ref PrevNetworkState))
+            if (!IsServer)
             {
-                ApplyNetworkState(ReplNetworkState.Value);
+                var oldStateDirtyInfo = UpdateNetworkStateCheckDirtyWithInfo(ref PrevNetworkState, 0);
+                if (oldStateDirtyInfo.isDirty && !oldStateDirtyInfo.isRotationDirty)
+                {
+                    // ignoring rotation dirty since quaternions will mess with euler angles, making this impossible to determine if the change to a single axis comes
+                    // from an unauthorized transform change or euler to quaternion conversion artifacts.
```

Comment:
if this is the reason why you're changing `UpdateNetworkState` stuff above, you could just leave things as they are and check for `HasRotAngleX || HasRotAngleY || HasRotAngleZ` because they're operating on the underlying `Bitset` and they'll be changed/updated after `UpdateNetworkState` call — you don't need "extra info", they're already there and set.